Skip PodGenerator import for deserialization when no k8s installed

potiuk opened the pull request:
The #56692 introduced optimization for PodGenerator imports - but there was a problem that when deserializing Pod it failed when no k8s classes were loaded - but it really is not optimisation but failure - nothing actually prevents us from importing the k8s classes and we actually **have to** do it in case we want to deserialize serialized Pod.

<!--
 Licensed to the Apache Software Foundation (ASF) under one
 or more contributor license agreements.  See the NOTICE file
 distributed with this work for additional information
 regarding copyright ownership.  The ASF licenses this file
 to you under the Apache License, Version 2.0 (the
 "License"); you may not use this file except in compliance
 with the License.  You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing,
 software distributed under the License is distributed on an
 "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 KIND, either express or implied.  See the License for the
 specific language governing permissions and limitations
 under the License.
 -->

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->



<!-- Please keep an empty line above the dashes. -->
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [airflow-core/newsfragments](https://github.com/apache/airflow/tree/main/airflow-core/newsfragments).

kaxil commented on airflow-core/src/airflow/serialization/serialized_objects.py:
```diff
@@ -925,8 +925,11 @@ def deserialize(cls, encoded_var: Any) -> Any:
+            # because if we need to deserialize Pod, we MUST have PodGenerator imported
+            _has_kubernetes(force=True)
+            from airflow.providers.cncf.kubernetes.pod_generator import PodGenerator
+
```
@potiuk This is a wrong fix! and will result in Scheduler or API-server crashing if k8s library was uninstalled/not-present anymore with weird errors

kaxil replied:
```suggestion
            if not _has_kubernetes(force=True):
                raise RuntimeError("Cannot deserialize POD objects without kubernetes libraries installed!")

```

kaxil replied:
It will already be imported by `_has_kubernetes` if k8s provider and library is installed after your "force" flag to recheck

potiuk replied:
I see - but Right now it crashes if the other side has no k8s libraries installed (see main failures) - because it cannot deserialize the objects - so I guess we should simply silently ignore it

potiuk replied:
Should be beter now.

kaxil replied:
The `force=True` alone will be enough without any changes. I had pushed a fix on your PR :) before your force-pushed it too :)

potiuk committed:
Skip PodGenerator import for deserialization when no k8s installed

The #56692 introduced optimization for PodGenerator imports - but there
was a problem that when deserializing Pod it failed when no k8s classes
were loaded - but it really is not optimisation but failure - nothing
actually prevents us from importing the k8s classes and we actually have
to do it in case we want to deserialize serialized Pod.  # Please enter
the commit message for your changes. Lines starting

kaxil committed:
fixup! Skip PodGenerator import for deserialization when no k8s installed

kaxil committed:
fixup! fixup! Skip PodGenerator import for deserialization when no k8s installed
